app/components/retriable_fast_image/index.tsx

```diff
@@ -0,0 +1,41 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+import React, {PureComponent} from 'react';
+import FastImage, {FastImageProps} from 'react-native-fast-image';
+
+export const FAST_IMAGE_MAX_RETRIES = 3;
+
+type RetriableFastImageProps = FastImageProps & {
+    id: string
```

Comment:
Where does the 'props' onError comes from ? If it is from FastImage, that's fine; else we'll need to add it to the RetriableFastImageProps.